adapters/core2sn/class.go

```diff
@@ -28,9 +28,12 @@ func AdaptCasmClass(coreCasmClass *core.CasmClass) starknet.CasmClass {
 	feederCompiledClass.CompilerVersion = coreCasmClass.CompilerVersion
 	feederCompiledClass.Hints = coreCasmClass.Hints
 	feederCompiledClass.Prime = utils.ToHex(coreCasmClass.Prime)
-	feederCompiledClass.BytecodeSegmentLengths = AdaptSegmentLengths(
-		coreCasmClass.BytecodeSegmentLengths,
-	)
+
+	if coreCasmClass.BytecodeSegmentLengths.Length != 0 ||
```

Comment:
If `BytecodeSegmentLengths` can be `nil`, is it correct to store it as value instead of pointer?